Functions/FirebaseFunctions/FIRFunctions.m

```diff
@@ -75,6 +99,12 @@ + (instancetype)functionsForApp:(FIRApp *)app region:(NSString *)region {
 }
 
 - (instancetype)initWithApp:(FIRApp *)app region:(NSString *)region {
+  return [self initWithApp:app region:region auth:FIR_COMPONENT(FIRAuthInterop, _app.container)];
+}
+
+- (instancetype)initWithApp:(FIRApp *)app
```

Comment:
Is this the `NS_DESIGNATED_INITIALIZER`? If so, please mark it as such.

Now that we only use `app` to get the project ID, can we change this to `initWithProjectID:(NSString *)projectID` and keep that in an ivar instead of the app? Then we could get rid of `FUNFakeApp` completely.

Reply:
I like this! The fewer references to app instances there are, the better.